File: api-spring-boot/etudiant-api/src/main/java/com/example/etudiant_api/entity/Etudiant.java
package com.example.etudiant_api.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.time.Period;

@Entity
@Table(name = "etudiants")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Etudiant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String email;
    private int anneePremiereInscription;

    @Column(unique = true, nullable = false)
    private String cin;

    @Column(nullable = false)
    private String nom;

    @Column(name = "date_naissance", nullable = false)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dateNaissance;

    @ManyToOne
    @JoinColumn(name = "departement_id")
    private Departement departement;

    @JsonIgnore
    public int age() {
        return Period.between(this.dateNaissance, LocalDate.now()).getYears();
    }
}

File: mobile_app/lib/models/etudiant.dart
class Etudiant {
  final int id;
  final String cin;
  final String nom;
  final String dateNaissance;
  final String? email;
  final int anneePremiereInscription;
  final int? departementId;
  final String? departementNom;

  Etudiant({
    required this.id,
    required this.cin,
    required this.nom,
    required this.dateNaissance,
    this.email,
    required this.anneePremiereInscription,
    this.departementId,
    this.departementNom,
  });

  factory Etudiant.fromJson(Map<String, dynamic> json) {
    return Etudiant(
      id: json['id'] ?? 0,
      cin: json['cin'] ?? '',
      nom: json['nom'] ?? '',
      dateNaissance: json['dateNaissance'] ?? '',
      email: json['email'],
      anneePremiereInscription: json['anneePremiereInscription'] ?? 0,
      departementId: json['departementId'],
      departementNom: json['departementNom'],
    );
  }
}